pkg/common/policy/policy.go

```diff
@@ -0,0 +1,170 @@
+package policy
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"os"
+
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/open-policy-agent/opa/storage"
+	"github.com/open-policy-agent/opa/storage/inmem"
+	"github.com/open-policy-agent/opa/util"
+)
+
+// Engine drives policy management.
+type Engine struct {
+	rego rego.PartialResult
+}
+
+type EngineConfig struct {
+	FileProvider *FileProviderConfig `hcl:"file_provider"`
+}
+
+type FileProviderConfig struct {
+	PolicyPath      string `hcl:"policy_path"`
+	PermissionsPath string `hcl:"permissions_path"`
+}
+
+// Input represents context associated with an access request.
+type Input struct {
+	// Caller is the authenticated identity of the actor making a request.
+	Caller string `json:"caller"`
+
+	// FullMethod is the fully-qualified name of the proto rpc service method.
+	FullMethod string `json:"full_method"`
+
+	// Req represents data received from the request body. It MUST be a
+	// protobuf request object with fields that are serializable as JSON,
+	// since they will be used in policy definitions.
+	Req interface{} `json:"req"`
+}
+
+type Result struct {
+	Allow             bool `json:"allow"`
+	AllowIfAdmin      bool `json:"allow_if_admin"`
+	AllowIfLocal      bool `json:"allow_if_local"`
+	AllowIfDownstream bool `json:"allow_if_downstream"`
+	AllowIfAgent      bool `json:"allow_if_agent"`
+}
+
+// NewEngine returns a new policy engine. Or nil if no
+// config is provided.
+func NewEngine(cfg *EngineConfig) (*Engine, error) {
+	if cfg == nil || cfg.FileProvider == nil {
+		return nil, nil
+	}
```

Comment:
Returning `nil` from a function named `NewEngine` is a bit weird IMO. I'd rather see this function not called if config is empty. Also, IMO, if the config is provided but doesn't have a valid provider set (i.e. FileProvider) that should probably be a failure instead of a silent ignore?